components/Feature/SharePlan/index.js

```diff
@@ -0,0 +1,100 @@
+import { useState } from 'react';
+import Table, {
+  TableHead,
+  TableRow,
+  TableHeader,
+  TableBody,
+  TableData
+} from 'components/Table';
+import { Button, Checkbox } from 'components/Form';
+import Heading from 'components/Heading';
+import css from './index.module.scss';
+import ShareStatus from './ShareStatus';
+
+const SharePlan = ({ error, plan, onPlanShared }) => {
+  const [selectedContact, setSelectedContact] = useState({});
+  const [disableShare, setDisableShare] = useState(true);
+
+  const handleSelectEmail = e => {
+    if (!e.target.value) return;
+    const contact = selectedContact;
+    e.target.checked ? (contact.email = e.target.value) : delete contact.email;
+    setSelectedContact(contact);
+    setDisableShare(Object.keys(selectedContact).length === 0);
```

Comment:
We could compute this inside of render instead of storing it in state, especially since it's not super expensive to calculate, that way we don't need to worry about keeping it up-to-date ourselves.